gulp/full.js

```diff
@@ -1,3 +1,15 @@
 var gulp = require('gulp');
+var gulpSequence = require('gulp-sequence');
 
-gulp.task('full', ['build', 'tslint', 'all-tests']);
+var isMaster = require('./util/is-master');
+var isTagAdded = require('./util/is-tag-added');
+
+gulp.task('build-and-tests', ['build', 'tslint', 'all-tests']);
+gulp.task('build-and-tests-and-publish', gulpSequence('build-and-tests', 'publish'));
+
+if (isMaster() && isTagAdded()) {
```

Comment:
Hmm, maybe. In the future are you going to place all logic for publishing to appveyor.yml? cdnjs, for example. Partially It is a right thing I think, but it makes you dependent on CI system unlike of gulp tasks infrastructure which can be used on any CI.